pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -706,8 +706,15 @@ func cleanStatus(machine *v3.Node) {
 	machine.Status.InternalNodeStatus.Images = nil
 	machine.Status.InternalNodeStatus.Conditions = conditions
 
-	delete(machine.Status.NodeAnnotations, podresources.RequestsAnnotation)
-	delete(machine.Status.NodeAnnotations, podresources.LimitsAnnotation)
+	annoMap := make(map[string]string, len(machine.Status.NodeAnnotations))
+	for key, val := range machine.Status.NodeAnnotations {
+		if key == podresources.LimitsAnnotation || key == podresources.RequestsAnnotation {
+			continue
+		}
+		annoMap[key] = val
+	}
+
+	machine.Status.NodeAnnotations = annoMap
```

Comment:
I think this is not needed, `machine.Status.NodeAnnotations` is already made a deepcopy in `convertNodeToMachine`

Reply:
The issue is that someone could use the `cleanStatus` function outside of the `convertNodeToMachine` and we would be back in the same position. Just trying to protect against this from happening in the future.

Reply:
What about `machine.Status.Conditions`? It's also modified in this function.

Reply:
Conditions is an slice. I don't think slices have the same problem with iteration/write panics.